modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v10_4_1/ObjectDefinitionUpgradeProcess.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.internal.upgrade.v10_4_1;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.dao.orm.common.SQLTransformer;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Alberto Sousa
+ */
+public class ObjectDefinitionUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				SQLTransformer.transform(
+					StringBundler.concat(
+						"SELECT cn.classNameId from ClassName_ cn where ",
+						"cn.value like '%objectDefinition#%' and cn.value not ",
+						"in (SELECT od.className from ObjectDefinition od ",
+						"where od.className like '%objectDefinition#%')")));
```

Comment:
I researched and found out that NOT IN can be less efficient for large datasets. So, what about using `left join`?
```
SELECT ClassName_.classNameId FROM ClassName_ 
LEFT JOIN ObjectDefinition ON ClassName_.value = ObjectDefinition.className
WHERE ClassName_.value LIKE '%objectDefinition#%'
AND ObjectDefinition.className IS NULL;
```

Reply:
Prefer to use the full name of the `tableName` instead of a short name like `cn` or `od`.

Reply:
Write the whole query in **lowercase**.